### PR TITLE
feat: fix profile link cell in the settings [OUTDATED] - ACC - 264

### PR DESCRIPTION
**Pull request by BrzozowskiPawel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

In this PR I have fixed the profile link cell.

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by MarcinR — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 1]
```diff
@@ -404,6 +404,8 @@ final class SettingsStaticTextTableCell: SettingsTableCell {
         super.setup()
         cellNameLabel.numberOfLines = 0
         cellNameLabel.textAlignment = .justified
+        backgroundView?.backgroundColor = SemanticColors.View.backgroundDefault
+        removeBorder()
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:408]: I don't think this is a good way to add a border and then remove it. And to use `accessibilityIdentifier = "border"` to find the view is extremely hacky.

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 2]
```diff
@@ -404,6 +404,8 @@ final class SettingsStaticTextTableCell: SettingsTableCell {
         super.setup()
         cellNameLabel.numberOfLines = 0
         cellNameLabel.textAlignment = .justified
+        backgroundView?.backgroundColor = SemanticColors.View.backgroundDefault
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:407]: Why not `backgroundColor = SemanticColors.View.backgroundDefault `?

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift (hunk 1 above)
Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:408]: In my opinion by this approach:
- the border have the accisibilityIdentifier 
- we can easily identify the specific view from the view hierarchy and delete it when it's not necessary. By default all cells have it.

That's why I think this approach is good.

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsProfileLinkCellDescriptor.swift [hunk 3]
```diff
@@ -24,7 +24,12 @@ class SettingsProfileLinkCellDescriptor: SettingsCellDescriptorType {
     // MARK: - Configuration
 
     func featureCell(_ cell: SettingsCellType) {
-        (cell as! SettingsProfileLinkCell).label.attributedText = title && .lineSpacing(8)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsProfileLinkCellDescriptor.swift:27]: You don't need to make these changes

Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsProfileLinkCellDescriptor.swift:27]: This is how currently (on the develop) the app looks like: 
image

And this is how it's specified in Figma:
image


That's why I have added the underline.

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsProfileLinkCellDescriptor.swift:27]: Screenshot 2022-09-06 at 14 56 26

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsProfileLinkCellDescriptor.swift:27]: I think we can keep the old logic

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift (hunk 2 above)
Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:407]: At first, I thought the same. 
Here is why I coded it as above:
image

And in SettingsTableCell (our super class) in line 189 we have:
backgroundView = UIView()

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:407]: Do we really need it?
`backgroundView = UIView()`

**Comment by BrzozowskiPawel** [comment]
@KaterinaWire I have reverted the underlined text for the link.

**Comment by BrzozowskiPawel** [comment]
@KaterinaWire 
I made a new commit with the redesign. 
Please take a look at the current approach and decide if it's proper. 

The first 3 snapshots were failing so I fixed them to match the overall style.

File: Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift [hunk 4]
```diff
@@ -277,6 +277,12 @@ final class SettingsTableViewController: SettingsBaseTableViewController {
         navigationItem.titleView = titleLabel
     }
 
+    private func addBorderFor(for cell: SettingsTableCell, descriptor cellDescriptor: SettingsCellDescriptorType) {
+        if  type(of: cellDescriptor).cellType.reuseIdentifier == SettingsStaticTextTableCell.reuseIdentifier ||
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift:281]: If we decide to add a new item to the settings and use the `SettingsStaticTextTableCell`, then this cell will never have a border. It means the code will not be reused.
I suggest to create a new cell which will have 2 labels: title (Profile link ) and link. With this approach, we don't need to remove any borders or update existing logic.

**Pull request closed without merging**